xonsh/commands_cache.py

```diff
@@ -75,41 +75,58 @@ def remove_dups(p):
                 ret.append(e)
         return ret
 
+    @staticmethod
+    def get_key(cmd):
+        return cmd.upper() if ON_WINDOWS else cmd
```

Comment:
I don't think we need a new public API for this

Reply:
reverted